src/components/OrgUnitDimension/OrgUnitDimension.js

```diff
@@ -1,276 +1,336 @@
-        } else {
-            this.props.onSelect({
-                dimensionId: DIMENSION_ID_ORGUNIT,
-                items: [
-                    ...selected,
-                    {
-                        ...orgUnit,
-                        name: orgUnit.name || orgUnit.displayName,
-                    },
-                ],
-            })
-        }
-    }
-
-    handleUserOrgUnitClick = (event, checked) => {
-        if (checked) {
-            if (!this.state.selected.length) {
-                this.setState({
-                    selected: this.props.ouItems.slice(),
-                })
+    const getSummary = () => {
+        let summary
+        if (selected.length) {
+            const numberOfOrgUnits = selected.filter(
+                item =>
+                    !DYNAMIC_ORG_UNITS.includes(item.id) &&
+                    !ouIdHelper.hasLevelPrefix(item.id) &&
+                    !ouIdHelper.hasGroupPrefix(item.id)
+            ).length
+
+            const numberOfLevels = selected.filter(item =>
+                ouIdHelper.hasLevelPrefix(item.id)
+            ).length
+            const numberOfGroups = selected.filter(item =>
+                ouIdHelper.hasGroupPrefix(item.id)
+            ).length
+            const userOrgUnits = selected.filter(item =>
+                DYNAMIC_ORG_UNITS.includes(item.id)
+            )
+
+            const parts = []
+
+            if (numberOfOrgUnits) {
+                parts.push(
+                    i18n.t('{{count}} org units', {
+                        count: numberOfOrgUnits,
+                        defaultValue: '{{count}} org unit',
+                        defaultValue_plural: '{{count}} org units',
+                    })
+                )
             }
-
-            this.props.onSelect({
-                dimensionId: DIMENSION_ID_ORGUNIT,
-                items: [
-                    ...this.props.ouItems.filter(ou =>
-                        this.userOrgUnitIds.includes(ou.id)
-                    ),
-                    userOrgUnits.find(ou => ou.id === event.target.name),
-                ],
+            if (numberOfLevels) {
+                parts.push(
+                    i18n.t('{{count}} levels', {
+                        count: numberOfLevels,
+                        defaultValue: '{{count}} level',
+                        defaultValue_plural: '{{count}} levels',
+                    })
+                )
+            }
+            if (numberOfGroups) {
+                parts.push(
+                    i18n.t('{{count}} groups', {
+                        count: numberOfGroups,
+                        defaultValue: '{{count}} group',
+                        defaultValue_plural: '{{count}} groups',
+                    })
+                )
+            }
+            userOrgUnits.forEach(orgUnit => {
+                parts.push(orgUnit.name || orgUnit.displayName)
             })
+            summary = i18n.t('Selected: ') + parts.join(', ')
```

Comment:
I have a similar thing for the sharing summary.
I don't know what is the best solution, as for some languages this "hardcoded" comma-separated list of text can be a problem.
I don't have a better suggestion tho...

Reply:
I had a long chat with Joe about this, because it started as "A," + "B," + "C"+ " selected, which is bad because they're all part of the same sentence. However with the "Selected: " + "parts", at least the sentence is split up. I could do the same as you do though, inject the comma list with interpolation instead of just concating the string.

Reply:
Curtesy of @mediremi :
```js
const formatList = items => {
    // Wrap Intl.ListFormat in try/catch as DHIS2 locales are not always ISO 639 compliant
    try {
        const formatter = new Intl.ListFormat(i18n.language, {
            style: 'long',
            type: 'conjunction',
        })
        return formatter.format(items)
    } catch (error) {
        return items.join(', ')
    }
}
```
([Source](https://github.com/dhis2/user-app/blob/8b323c366b9e18421856f47e1ba76a4d2ad5d3db/src/containers/UserList/OrganisationUnitFilter.js#L10-L21))

Reply:
Looks good, added 👍  Thanks!